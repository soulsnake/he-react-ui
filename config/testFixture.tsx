import * as React from 'react';
import createTestContext from 'react-cosmos-test/enzyme';
import toJSON from 'enzyme-to-json';

interface Fixture {
  component: React.ComponentType<any>;
  name?: string;
}

interface EnzymeElement {
  node: {
    nodeType: string;
    props: {
      children?: null;
    };
  };
  children: null | (EnzymeElement | null)[];
}

function sanitiseSubtree(
  element: EnzymeElement | null,
  compositeSeenPreviously: boolean,
): EnzymeElement | null {
  if (!element) return element;
  const { node, children } = element;
  if (!node) return element;
  const { nodeType, props } = node;
  const isComposite = nodeType !== 'host';
  const hasChildrenProp = Boolean(props.children);
  const shouldIncludeChildren =
    !isComposite || !compositeSeenPreviously || hasChildrenProp;
  const compositeSeen = compositeSeenPreviously || isComposite;
  const cleanedChildren: (EnzymeElement | null)[] | null = shouldIncludeChildren
    ? children &&
      children.map((it: EnzymeElement | null) =>
        sanitiseSubtree(it, compositeSeen),
      )
    : [];

  return {
    ...element,
    children: cleanedChildren,
  };
}

function shallowSerialise(enzymeTree: any): EnzymeElement | null {
  return sanitiseSubtree((toJSON(enzymeTree) as any) as EnzymeElement, false);
}

function testOneFixture(fixture: Fixture, variant: string = 'a') {
  const { mount, unmount, getWrapper } = createTestContext({ fixture });
  const { component } = fixture;
  const { displayName, name } = component as any;

  test(`<${displayName || name} /> rendered correctly with ${fixture.name ||
    variant} fixture`, () => {
    mount();
    try {
      expect(shallowSerialise(getWrapper())).toMatchSnapshot();
    } finally {
      unmount();
    }
  });
}

// TODO: use real type guard functions to discriminate one fixture vs an array of them

export default function testAllFixtures(fixtures: any, variant?: string) {
  if (!('forEach' in fixtures)) {
    testOneFixture(fixtures as Fixture, variant);
  } else if (typeof fixtures.forEach === 'function') {
    (fixtures as Fixture[]).forEach(it => testOneFixture(it));
  }
}
