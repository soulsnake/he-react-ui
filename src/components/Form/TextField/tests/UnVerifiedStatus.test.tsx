import fixture from '../fixtures/UnVerifiedStatus.fixture';
import testFixture from '../../../../../config/testFixture';

testFixture(fixture);