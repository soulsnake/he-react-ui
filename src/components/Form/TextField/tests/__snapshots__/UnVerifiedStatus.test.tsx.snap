// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextField /> rendered correctly with UnVerifiedStatus fixture 1`] = `
<TextField
  disabled={false}
  id="demo"
  inline={false}
  isValid={true}
  isVerified={false}
  label="Mobile"
  marker="verify"
  markerTooltip={
    <UNDEFINED>
      <mockConstructor
        className="editIcon"
        name="Edit"
        shadow={false}
      />
      <mockConstructor
        h1={false}
        h2={false}
        h3={true}
        h4={false}
        nav={false}
      >
        Sample Popup
      </mockConstructor>
      <Paragraph
        className="paragraph"
      >
        Sample content
      </Paragraph>
    </UNDEFINED>
  }
  name="demo"
  onBlur={[Function]}
  onChange={[Function]}
  onFocus={[Function]}
  onMarkerClick={[Function]}
  value=""
>
  <div
    className="outer hasMarker"
  >
    <div
      className="block"
    >
      <input
        className="input"
        disabled={false}
        id="demo"
        name="demo"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="text"
        value=""
      />
      <div
        className="helper"
      >
        <Popover
          content={
            <UNDEFINED>
              <mockConstructor
                className="editIcon"
                name="Edit"
                shadow={false}
              />
              <mockConstructor
                h1={false}
                h2={false}
                h3={true}
                h4={false}
                nav={false}
              >
                Sample Popup
              </mockConstructor>
              <Paragraph
                className="paragraph"
              >
                Sample content
              </Paragraph>
            </UNDEFINED>
          }
          light={true}
          preferRight={true}
        >
          <Tooltip
            animateFill={true}
            animation="shift"
            arrow={false}
            arrowSize="regular"
            className=""
            delay={100}
            disabled={false}
            distance={10}
            duration={375}
            followCursor={false}
            hideDelay={300}
            hideDuration={375}
            hideOnClick={true}
            html={
              <PopoverDisplay
                className={undefined}
                light={true}
                tooltip={undefined}
              >
                <UNDEFINED>
                  <mockConstructor
                    className="editIcon"
                    name="Edit"
                    shadow={false}
                  />
                  <mockConstructor
                    h1={false}
                    h2={false}
                    h3={true}
                    h4={false}
                    nav={false}
                  >
                    Sample Popup
                  </mockConstructor>
                  <Paragraph
                    className="paragraph"
                  >
                    Sample content
                  </Paragraph>
                </UNDEFINED>
              </PopoverDisplay>
            }
            inertia={false}
            interactive={true}
            interactiveBorder={16}
            multiple={false}
            offset={0}
            onHidden={[Function]}
            onHide={[Function]}
            onRequestClose={[Function]}
            onShow={[Function]}
            onShown={[Function]}
            popperOptions={Object {}}
            position="bottom-end"
            size="regular"
            sticky={false}
            stickyDuration={200}
            style={Object {}}
            theme="dark"
            touchHold={false}
            trigger="mouseenter"
            unmountHTMLWhenHide={false}
          >
            <div
              className=""
              style={
                Object {
                  "display": "inline",
                }
              }
            >
              <div
                className="verifiedTag unVerified"
                onClick={[Function]}
              >
                Verify
              </div>
            </div>
          </Tooltip>
        </Popover>
      </div>
      <label
        className="label"
        htmlFor="demo"
      >
        Mobile
      </label>
    </div>
  </div>
</TextField>
`;
